core/src/main/java/org/elasticsearch/common/xcontent/ConstructingObjectParser.java

```diff
@@ -295,6 +295,10 @@ public Value parse(XContentParser parser, Context context) throws IOException {
         }
     }
 
+    public String getName() {
```

Comment:
+1, I think I did this in some WIP branch as well the other day. Didn't get back to it though.